easybuild/framework/easyblock.py

```diff
@@ -4198,6 +4207,98 @@ def print_dry_run_note(loc, silent=True):
     dry_run_msg(msg, silent=silent)
 
 
+def persist_failed_compilation_log_and_artifacts(build_successful, application_log, silent, app, easyconfig):
+    persistent_dirs = {}
+
+    def get_dir(path, dirname):
+        nonlocal persistent_dirs
+        dirpath = persistent_dirs.get(dirname)
+        if dirpath is None:
+            dirpath = create_unused_path(path)
+            persistent_dirs[dirname] = dirpath
+        return persistent_dirs[dirname]
+
+    def get_artifact_dir(path):
+        return get_dir(path, 'artifact')
+
+    def get_log_dir(path):
+        return get_dir(path, 'log')
+
+    def initialize_persistent_dir_and_replicate_if_paths_distinct(replication_operation,
+                                                                  get_directory,
+                                                                  source_path,
+                                                                  destination_base_path,
+                                                                  destination_relative_path):
+        abs_source_path = os.path.abspath(source_path)
+        abs_destination_base_path = os.path.abspath(destination_base_path)
+        abs_destination_path = os.path.join(abs_destination_base_path, destination_relative_path)
```

Comment:
It would be easier to understand if those 2 were 1 argument in the first place. Had to search whether either was used again. (Too many variables make it harder to reason in general.)